execution.py

```diff
@@ -4,43 +4,115 @@
 import threading
 import heapq
 import traceback
+from enum import Enum
 import inspect
 from typing import List, Literal, NamedTuple, Optional
 
 import torch
 import nodes
 
 import comfy.model_management
+import comfy.graph_utils
+from comfy.graph import get_input_info, ExecutionList, DynamicPrompt, ExecutionBlocker
+from comfy.graph_utils import is_link, GraphBuilder
+from comfy.caching import HierarchicalCache, LRUCache, CacheKeySetInputSignature, CacheKeySetInputSignatureWithID, CacheKeySetID
+from comfy.cli_args import args
+
+class ExecutionResult(Enum):
+    SUCCESS = 0
+    FAILURE = 1
+    SLEEPING = 2
+
+class IsChangedCache:
+    def __init__(self, dynprompt, outputs_cache):
+        self.dynprompt = dynprompt
+        self.outputs_cache = outputs_cache
+        self.is_changed = {}
+
+    def get(self, node_id):
+        if node_id not in self.is_changed:
+            node = self.dynprompt.get_node(node_id)
+            class_type = node["class_type"]
+            class_def = nodes.NODE_CLASS_MAPPINGS[class_type]
+            if hasattr(class_def, "IS_CHANGED"):
+                if "is_changed" in node:
```

Comment:
Oh, I see.

Technically, people submitting prompts can include their own `is_changed` values  on nodes that will be honored instead of evaluating the `IS_CHANGED` function. I'm not entirely sure what the use-case is for that, but since it's existing functionality, I was hesitant to remove it without having a good reason to do so. I'll bring it up in Matrix and see what people think.